parlai/core/torch_generator_agent.py

```diff
@@ -1059,6 +1065,36 @@ def get_prefix_tokens(self, batch: Batch) -> Optional[torch.LongTensor]:
         """
         return None
 
+    def get_opt_prefix(self, batch: Batch) -> Optional[torch.LongTensor]:
+        """
+        Set prefix tokens to seed decoding at generation time.
+
+        By default, we do not utilize prefix tokens, but this is
+        left overridable by child classes.
+
+        Returned tensor should be of dimension bsz x len(prefix)
+        """
+        prefix_toks = self.opt.get("opt_prefix_tokens", [])
+        if len(prefix_toks) == 0:
+            return None
+        bsz = batch.batchsize
+        dev = batch.text_vec.device
+        prefix_toks_batch = [prefix_toks for _ in range(bsz)]
+        return torch.LongTensor(prefix_toks_batch).to(dev)
+
+    def get_all_prefix(self, batch: Batch) -> Optional[torch.LongTensor]:
```

Comment:
IDK, this looks to make the code a bit patchy for general use. Maybe we should keep this in a branch for our own project and not merge it into ParlAI, until we have a cleaner solution. In other words, just keep this branch updated (with rebase) with the main branch and use it for your project.